### DISCUSSION TITLE
Is there a way to call forbidden functions with in a module.

### DISCUSSION THREAD

**Discussion by leigh-pointer** [oqtane/oqtane.framework#2151] [Q&A]
I'm trying to call this function when I have no user logged in.
`await UserRoleService.GetUserRolesAsync(PageState.Site.SiteId);`

Debugging through, the result from the request is "forbidden"

**Comment by sbwalker** [comment]
The GetUserRoles API method retrieves the list of security roles defined for the specified site. This information is sensitive and should not be available to anonymous users - which is why you are getting a Forbidden error in your client application. If you have a scenario where an anonymous user needs to access this information you would need to create your own API method in your module and within that method it can call the repository and get the user roles. However I would highly discourage you from simply sending this info to the client - I would evaluate why you need this information and try to focus the workload on the server so that you do not leak sensitive information publicly.

**Reply by leigh-pointer** [reply]
@sbwalker on reflection I am going to refactor my model, thanks Shaun.
